apis/supervisor/idp/v1alpha1/types_githubidentityprovider.go.tmpl

```diff
@@ -0,0 +1,219 @@
+// Copyright 2024 the Pinniped contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type GitHubIdentityProviderPhase string
+
+const (
+	// GitHubPhasePending is the default phase for newly-created GitHubIdentityProvider resources.
+	GitHubPhasePending GitHubIdentityProviderPhase = "Pending"
+
+	// GitHubPhaseReady is the phase for an GitHubIdentityProvider resource in a healthy state.
+	GitHubPhaseReady GitHubIdentityProviderPhase = "Ready"
+
+	// GitHubPhaseError is the phase for an GitHubIdentityProvider in an unhealthy state.
+	GitHubPhaseError GitHubIdentityProviderPhase = "Error"
+)
+
+type GitHubRestrictGitHubUsersToAllowedOrganizations string
+
+const (
+	// GitHubRestrictGitHubUsersToAllowedOrganizationsYes means only those users with membership in the listed GitHub
+	// organizations are allowed to log in.
+	GitHubRestrictGitHubUsersToAllowedOrganizationsYes GitHubRestrictGitHubUsersToAllowedOrganizations = "Yes"
+
+	// GitHubRestrictGitHubUsersToAllowedOrganizationsNo means any GitHub user is allowed to log in using this identity provider,
+	// regardless of their organization membership or lack thereof.
```

Comment:
Per our discussions about:
- Naming
- Avoiding Booleans & "Yes","No" values.

```yaml
spec:
  restrictUsers: "allowed-organizations" # enum values, can drop "github" (since this is the github CRD anyway) and make some of the name the configuration value.
```

Reply:
makes sense.